pkg/operator/controller/ingress/controller.go

```diff
@@ -385,6 +388,17 @@ func setDefaultPublishingStrategy(ic *operatorv1.IngressController, infraConfig
 		if effectiveStrategy.HostNetwork == nil {
 			effectiveStrategy.HostNetwork = &operatorv1.HostNetworkStrategy{}
 		}
+		// explicitly set the default ports if some of them are omitted
+		if effectiveStrategy.HostNetwork.HTTPPort == 0 {
```

Comment:
/hold

It's not clear why we have defaults here and in the API.
If the API allowed 0 we would have defaults in a single place. How would we know to change the code here if we change the API's current defaults?

Reply:
/hold cancel

I no longer have any concerns about defaults, particularly since https://github.com/openshift/api/pull/1175 merged.